Feature/wrap logging

Burke-Daniel opened the pull request:
This PR contains a Log class which wraps the logging functionality of easylogging++ to provide a core logger instance to be used within the engine for logging, and a client logger instance to be used by developers utilizing the engine to log within their application.

One thing I have not implemented yet is the ability to set the log level (i.e. setting log level so that anything below the set log level will not be logged to the console). Easylogging++ is a bit weird in that it does not hierarchically log by default, but I feel like it would be pretty easy to add this functionality within the Log wrapper.

We should discuss and then subsequently implement if necessary.

Burke-Daniel committed:
Glob all files under Src in aquarius engine

All .cpp and .h files under the Src directory will automatically be added to the cmake Aquarius-Engine project

Burke-Daniel committed:
Update test class to use Aquarius logging

colton-smith commented on Aquarius/Src/Aquarius/Core/Log.h:
```diff
@@ -0,0 +1,37 @@
+// Core log macros
+#define AQ_CORE_TRACE(String, ...)		::Aquarius::Log::getCoreLogger()->trace(String, __VA_ARGS__);
+#define AQ_CORE_INFO(String, ...)		::Aquarius::Log::getCoreLogger()->info(String, __VA_ARGS__);
+#define AQ_CORE_WARNING(String, ...)	::Aquarius::Log::getCoreLogger()->warn(String, __VA_ARGS__);
```
This line should be indented like the others

colton-smith commented on Aquarius/Src/Aquarius/Core/Log.h:
```diff
@@ -0,0 +1,37 @@
+// Client log macros
+#define AQ_TRACE(String, ...)			::Aquarius::Log::getClientLogger()->trace(String, __VA_ARGS__);
+#define AQ_INFO(String, ...)			::Aquarius::Log::getClientLogger()->info(String, __VA_ARGS__);
+#define AQ_WARNING(String, ...)			::Aquarius::Log::getClientLogger()->warn(String, __VA_ARGS__);
```
Same with this one

liamjwarren replied:
Fix all ::Aquarius to be inline with each other.

liamjwarren commented on Aquarius/Src/Aquarius/Core/Log.h:
```diff
@@ -0,0 +1,37 @@
+// Core log macros
+#define AQ_CORE_TRACE(String, ...)		::Aquarius::Log::getCoreLogger()->trace(String, __VA_ARGS__);
+#define AQ_CORE_INFO(String, ...)		::Aquarius::Log::getCoreLogger()->info(String, __VA_ARGS__);
+#define AQ_CORE_WARNING(String, ...)	::Aquarius::Log::getCoreLogger()->warn(String, __VA_ARGS__);
```
Fix all ::Aquarius to be inline with each other.

Burke-Daniel replied:
Check this in your text editor, I believe they are inline, github tends to be funny when tabs and spaces are mixed together

Burke-Daniel replied:
Nvm fixed this by changing tabs to spaces, thanks visual studio for doing weird shit

colton-smith commented on Aquarius/Src/Aquarius/Core/Log.h:
```diff
@@ -0,0 +1,37 @@
+	private:
+		Log() = delete;
+
+		static el::Logger* s_clientLogger;
+		static el::Logger* s_coreLogger;
+	};
```
Presumably these are okay being a naked pointer, I'd say that their lifetime is managed elsewhere right? 

Also - whats up with the s_ prefix? Is it for indicating static members? Im cool with that if so

Burke-Daniel replied:
Yeah I think the lifetime is managed by easylogging++, seems like you can grab a pointer to these guys with `el::Loggers::getLogger` whenever you please.

Yup s_ indicates static class members

colton-smith reviewed (approved):
Looks good, gonna approve now, only thing I noticed was indentation due to the spaces/tabs like we discussed. Whether or not thats fixed, IDRC tbh, but looking clean in github is always nice, afterall - the hub is where people are gonna see it most I feel like.

Dir structure update is mint too!

Burke-Daniel committed:
Wrap easylogging++ log functionality

This wraps easylogging into client and core logger instances with specific output formatting
Fix indentation in Log.h
